test/core/expectations/compound_data/test_expectations_cursor.pkb

```diff
@@ -1041,5 +1041,516 @@ Rows: [ 2 differences ]%
+  end;
+  
+  procedure inlc_dif_cols_as_list
+  as
+    l_actual   sys_refcursor;
+    l_expected sys_refcursor;
+  begin
+    --Arrange
+    open l_actual   for select rownum as rn, 'TEST' as A_COLUMN  from dual a connect by level < 4;
+    open l_expected for select rownum as rn, 1 as A_COLUMN  from dual a connect by level < 4;
+    --Act
+    ut3.ut.expect(l_actual).to_equal(l_expected).include(ut3.ut_varchar2_list('RN'));
+    --Assert
+    ut.expect(expectations.failed_expectations_data()).to_be_empty();
+  end;
+
+  procedure inlc_exc_dif_cols_as_list
+  as
+    l_actual   sys_refcursor;
+    l_expected sys_refcursor;
+  begin
+    --Arrange
+    open l_actual   for select rownum as rn, 'TEST' as A_COLUMN  from dual a connect by level < 4;
+    open l_expected for select rownum as rn, 1 as A_COLUMN  from dual a connect by level < 4;
+    --Act
+    ut3.ut.expect(l_actual).to_equal(l_expected).include(ut3.ut_varchar2_list('RN')).exclude(ut3.ut_varchar2_list('A_COLUMN'));
+    --Assert
+    ut.expect(expectations.failed_expectations_data()).to_be_empty();
+  end;
+    
+  procedure compare_nested_tab_col_un is
+    l_actual   sys_refcursor;
+    l_expected sys_refcursor;
+  begin
+    --Arrange
+    open l_actual for select test_dummy_object( rownum, 'Something '||rownum, rownum) as colval
+      from dual connect by level <=2 order by rownum asc;
+
+    open l_expected for select test_dummy_object( rownum, 'Something '||rownum, rownum) as colval
+      from dual connect by level <=2 order by rownum desc;
+    
+    --Act
+    ut3.ut.expect(l_actual).to_equal(l_expected).unordered;
+    --Assert
+    ut.expect(expectations.failed_expectations_data()).to_be_empty();
+  end;
+  
+    procedure compare_nested_tab_col_jb is
+    l_actual   sys_refcursor;
+    l_expected sys_refcursor;
+  begin
+    --Arrange
+    open l_actual for select test_dummy_object( rownum, 'Something '||rownum, rownum) as colval
+      from dual connect by level <=2 order by rownum asc;
+
+    open l_expected for select test_dummy_object( rownum, 'Something '||rownum, rownum) as colval
+      from dual connect by level <=2 order by rownum desc;
+    
+    --Act
+    ut3.ut.expect(l_actual).to_equal(l_expected).join_by('COLVAL/ID');
+    --Assert
+    ut.expect(expectations.failed_expectations_data()).to_be_empty();
+  end;
+
+  procedure comp_nest_tab_col_un_fail is
+    l_actual   sys_refcursor;
+    l_expected sys_refcursor;
+  begin
+    --Arrange
+    open l_actual for select test_dummy_object( rownum, 'Something '||rownum, rownum) as colval
+      from dual connect by level <=2 order by rownum asc;
+
+    open l_expected for select test_dummy_object( rownum, 'Somethings '||rownum, rownum) as colval
+      from dual connect by level <=3 order by rownum desc;
+    
+    --Act
+    ut3.ut.expect(l_actual).to_equal(l_expected).unordered;
+    --Assert
+    ut.expect(expectations.failed_expectations_data()).to_be_not_null();
```

Comment:
What data will the user see in this case?
Can you test for an expected message?